pkg/monitoring/prometheusagent/secret.go

```diff
@@ -68,21 +103,15 @@ func (pas PrometheusAgentService) buildRemoteWriteSecret(
 	}, nil
 }
 
-func readRemoteWritePasswordFromSecret(secret corev1.Secret) (string, error) {
-	remoteWriteConfig := RemoteWriteConfig{}
-	err := yaml.Unmarshal(secret.Data["values"], &remoteWriteConfig)
+func readMimirAuthPasswordFromSecret(secret corev1.Secret) (string, error) {
+	var secretData string
+
+	err := yaml.Unmarshal(secret.Data["credentials"], &secretData)
 	if err != nil {
 		return "", errors.WithStack(err)
 	}
 
-	for _, rw := range remoteWriteConfig.PrometheusAgentConfig.RemoteWrite {
-		// We read the secret from the remote write configuration named `prometheus-meta-operator` only
-		// as this secret is generated per cluster.
-		// This will eventually be taken care of by the multi-tenancy contoller
-		if rw.Name == remoteWriteName {
-			return rw.Password, nil
-		}
-	}
+	password := strings.Split(secretData, ":")[1]
```

Comment:
Not prefixing with the username would remove this need

Reply:
yes, updated it alongside the removal of the username